### DISCUSSION TITLE
Auto Start Applications

### DISCUSSION THREAD

**Discussion by Veicm** [mylinuxforwork/dotfiles#407] [Q&A]
[start_programs.txt](https://github.com/user-attachments/files/17591175/start_programs.txt)

I want to start a few applications and adjust them automatically so I wrote this script but when I execute it I will get this issues 
`moveWindow: no window
moveWindow: no window
moveWindow: no window
moveWindow: no window`
and
`resizeWindow: no window
resizeWindow: no window
resizeWindow: no window
resizeWindow: no window`
and when I edit it to use directly the window name like in the [dispatch documentation](https://wiki.hyprland.org/Configuring/Dispatchers/) I will get the error massage `(` is an invalid syntax.
If anyone knows a different syntax to make the script run I would be very thankfully. And I'm sorry for my bad english.

**Comment by ap27** [comment]
@Veicm 
'Post festam'  I've made some progress I hope. 
[start_programs.txt](https://github.com/user-attachments/files/18022772/start_programs.txt)

**Reply by Veicm** [reply]
I just tried the program and added the right numbers for the variables and it worked, I don't get a error massage. But the windows are completely oversized and miss placed and when I look in `hyprctl clients` after I set everything up manually and after executing the script the numbers for size and position change completely.

**Reply by ap27** [reply]
Amateur card players say: Instead of intense guesswork a look at your partner's cards is worth a lot more. So, having ignored your window sizes and positions, what is your preferred arrangement? Some extra delays are also needed as it ran differently in autostart vs manual tests.
BTW for me (für einen Anfänger) a floating btop is a good example to use scratchpad
```
# Move/Switch to special workspace (scratchpad)
bind = $mainMod+Alt, S, movetoworkspacesilent, special
bind = $mainMod, S, togglespecialworkspace,
```

**Reply by Veicm** [reply]
Well, the function of a special workspace is very nice and I didn't know that till now but I don't see how this will help me with my Problem. Could you please explain that further? 
Thanks in advance.

**Reply by ap27** [reply]
btop needs 80x40 chars at least. I simple make its  window floating (manual testing Super+T, Super+Shift+arrows resizes, can be moved by Super+keep left mouse, Super+Alt+S sends to sratchpad. And Super+S later brings back to the same position if needed. Not a big invention. If there are too many floating windows you can clean them without memorizing where were they sent to silently.
